honeybee_radiance_command/gensky.py

```diff
@@ -0,0 +1,228 @@
+        if month and isinstance(month, int) and month in range(1, 13):
+            return month
+        else:
+            raise ValueError('The value of month needs to be an integer between'
+                             ' 1 and 12 inclusive. Instead got %.' % (month))
+
+    @staticmethod
+    def check_day(day):
+        """Validate the value of day."""
+        if day and isinstance(day, int) and day in range(1, 32):
+            return day
+        else:
+            raise ValueError('The value of day needs to be an integer between'
+                             ' 1 and 31 inclusive. Instead got %.' % (day))
+
+    @staticmethod
+    def check_time(time):
+        """Validate the value of time."""
+        if time and isinstance(time, float):
+            hour, minute = str(time).split('.')
+            if int(hour) in range(1, 25) and int(minute) in range(0, 60):
+                return time
+            else:
+                raise ValueError(
+                    'The value of time is not in the 24 hours format.'
+                )
+        else:
+            raise ValueError(
+                'The value of time must be a decimal representing hours and minutes.'
+            )
+
+    @staticmethod
+    def check_time_zone(time_zone):
+        """Validate the value of time-zone."""
+        time_zones = [
+            'YST', 'PST', 'MST', 'CST', 'EST', 'GMT', 'CET', 'EET', 'AST', 'GST', 'IST',
+            'JST', 'NZST', 'YDT', 'PDT', 'MDT', 'CDT', 'EDT', 'BST', 'CEST', 'EEST',
+            'ADT', 'GDT', 'IDT', 'JDT', 'NZDT']
+        if not time_zone:
+            return None
+        if time_zone and time_zone.upper() in time_zones:
+            return time_zone
+        else:
+            raise ValueError(
+                'Time zone must a three letter string from the following'
+                ' options %.' % (time_zones)
+            )
+
+    @staticmethod
+    def check_solar_time(solar_time):
+        """Validate value for solar time."""
+        if isinstance(solar_time, bool):
+            return solar_time
+        else:
+            raise ValueError('Solar time only accepts True or False as a value.')
+
+    def _gensky_time(self):
+        """Get time string to be used in gensky command."""
+        # Only month, day and time provided as arguments
+        if not (self.time_zone and self.solar_time):
+            gensky_input = ' '.join((str(self.month), str(self.day), str(self.time)))
+            return gensky_input
+        # Time zone is provided
+        elif self.time_zone and not self.solar_time:
+            gensky_time = ''.join((str(self.time), self.time_zone))
+            gensky_input = ' '.join((str(self.month), str(self.day), gensky_time))
+            return gensky_input
+        # Local solar time requested
+        elif self.solar_time and not self.time_zone:
+            gensky_time = ' '.join((str(self.month), str(self.day), str(self.time)))
+            gensky_input = ''.join(('+', gensky_time))
```

Comment:
What if we just use text formatting for most of these strange joins?

```suggestion
            gensky_input = '+%d %d %f' % (self.month, self.day, self.time)
```